app/models/newsletter_finder.rb

```diff
@@ -0,0 +1,31 @@
+class NewsletterFinder
+  include Enumerable
+
+  delegate :each, to: :newsletters_list
+
+  def initialize(type)
+    @type = type
+  end
+
+  private
+
+  def newsletters_list
+    @newsletters_list ||= newsletters_with_file_extension.map do |newsletter|
+      newsletter.gsub(/\.pdf/, "")
+    end
+  end
+
+  def newsletters_with_file_extension
+    unsorted_newsletters.sort do |date_1, date_2|
+      Date.parse("1-#{date_2}") <=> Date.parse("1-#{date_1}")
```

Comment:
![screen shot 2016-02-04 at 9 39 28 pm](https://cloud.githubusercontent.com/assets/11034063/12836530/da334068-cb87-11e5-8d59-0d318add53cd.png)
Without the 1- it does not accept the argument as a date. The files are saved as jun-15 for example and I mimicked this in the example in irb.